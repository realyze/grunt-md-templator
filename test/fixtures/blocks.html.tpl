<% _.forEach(children, function(main) { %>

<div id="<%= main.id %>">

  <%= main.content %>

  <% _.forEach(main.children, function(section) { %>

    <section id="<%= section.id %>">
      <%= section.body %>

      <% if (section.blocks.front && section.blocks.back) { %>

        <div class="I-am-the-front-block">
          <%= section.blocks.front.html %>
        </div>
        <div class="I-am-the-second-block">
          <%= section.blocks.back.html %>
        </div>

      <% } else { %>

        <% _.forEach(section.children, function(article) { %>
          <article id="<%= article.id %>">
            <%= article.content %>
          </article>
        <% }) %>

      <% } %>

    </section><% })
  %>

</div><%
}) %>
