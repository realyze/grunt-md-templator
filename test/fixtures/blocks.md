# Solutions

This is the Solutions noodle.

## Overview

some overview

--- front ---

This is block 1 of the Solutions noodle.

### Header 3
some 3rd level content within the `front` block

--- back ---

This is block 2 of the Solutions noodle and it lies within the `back` block.

## Web Apps

This is my rant about Web Apps which is not in any block because the 2nd level
header ended the block above (because it was a level-2-heading block).

Now here is some markdown just for fun:
 * with bullet points
 * oh yes!
 * with many bullet points!!!

--- front ---

### Block 2.1 header
This header and content is contained in the `front` block.

Yet another paragraph.

--- back ---

### Block 2.2 header
This header and content is contained in the `back` block.

## Last 2nd level header
...which is not contained in any block.
