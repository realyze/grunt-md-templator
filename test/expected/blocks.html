<div id="solutions">

    <h1>Solutions</h1>

    <p>This is the Solutions noodle.</p>



    <section id="overview">
        <p>some overview</p>



        <div class="I-am-the-front-block">
            <p>This is block 1 of the Solutions noodle.</p>
            <h3>Header 3</h3>

            <p>some 3rd level content within the <code>front</code> block</p>
        </div>
        <div class="I-am-the-second-block">
            <p>This is block 2 of the Solutions noodle and it lies within the <code>back</code> block.</p>
        </div>



    </section>

    <section id="web-apps">
        <p>This is my rant about Web Apps which is not in any block because the 2nd level header ended the block above (because it was a level-2-heading block).</p>

        <p>Now here is some markdown just for fun:
            <em> with bullet points
 </em> oh yes! * with many bullet points!!!</p>



        <div class="I-am-the-front-block">
            <h3>Block 2.1 header</h3>

            <p>This header and content is contained in the <code>front</code> block.</p>

            <p>Yet another paragraph.</p>
        </div>
        <div class="I-am-the-second-block">
            <h3>Block 2.2 header</h3>

            <p>This header and content is contained in the <code>back</code> block.</p>
        </div>



    </section>

    <section id="last-2nd-level-header">
        <p>...which is not contained in any block.</p>







    </section>

</div>
